src/mds/MDLog.cc

```diff
@@ -1435,9 +1435,24 @@ void MDLog::standby_trim_segments()
   bool removed_segment = false;
   while (have_any_segments()) {
     LogSegment *seg = get_oldest_segment();
-    if (seg->end > expire_pos)
+    dout(10) << " segment seq=" << seg->seq << " " << seg->offset <<
+      "~" << seg->end - seg->offset << dendl;
+
+    if (seg->end > journaler->get_read_pos()) {
```

Comment:
seg->end is set to journaler->get_read_pos() after reading each log event. How can this test be true?